### PR TITLE
UIU-3080 - Handle invalid image URLs when uploading profile photo

### PR DESCRIPTION
**Pull request by manvendra-s-rathore**
[UIU-3080](https://folio-org.atlassian.net/browse/UIU-3080) - Handle invalid image URLs when uploading profile photo.

## Screencast 
https://github.com/folio-org/ui-users/assets/105021655/62c758d5-199c-40e8-86bc-a522c610e8c1

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js [hunk 1]
```diff
@@ -23,37 +23,41 @@ const ExternalLinkModal = ({
...
     }
   }, [inputValue]);
 
-  const handleSave = () => {
-    onSave(inputValue);
+  const handleSave = async () => {
+    const isValidImgURL = await isAValidImageUrl(inputValue);
+    if (isValidImgURL) {
+      onSave(inputValue);
+    }
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js:43]: Image URL need not be validated once again as it will be already validated onBlur.
It can be saved in case of no error?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 * Trim input values and delete properties with empty string when user record save. Refs UIU-2049.
 * Update username field validation to trim leading and trailing spaces. Refs UIU-3099.
 * Fix "Total paid amount" value that set as "$NaN" on "Refund fee/fine" modal. Refs UIU-3094.
+* Handle invalid image URLs when uploading profile photo via External URL. Refs UIU-3080.
```
Comment (Terala-Priyanka) [inline CHANGELOG.md:11]: Could you reword this to something like "Validate image url provided as external url for user profile picture"?

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js [hunk 3]
```diff
@@ -23,37 +23,41 @@ const ExternalLinkModal = ({
...
   const handleInputChange = (e) => {
     previousInputValue.current = inputValue;
     setInputValue(e.target.value);
   };
 
-  const handleBlur = () => {
-    if (inputValue && !isAValidURL(inputValue)) {
-      setError(true);
+  const handleBlur = async () => {
+    if (!inputValue || !isAValidURL(inputValue)) {
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js:52]: This will not handle a case when end user clears the url and clicks somewhere outside the url text box and it will throw error as invalid url when url is actually empty.

File: src/components/util/util.js [hunk 4]
```diff
@@ -222,3 +222,14 @@ export const isAValidURL = (str) => {
   return URL.canParse(str);
 };
 
+export const isAValidImageUrl = async (url) => {
+  try {
+    const response = await fetch(url);
```
Comment (Terala-Priyanka) [inline src/components/util/util.js:227]: This is throwing CORS issue when tried with url like https://folio.org/wp-content/uploads/2023/08/folio-site-general-Illustration-social-image-1200.jpg

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]

**Comment by manvendra-s-rathore** [comment]
Hey @folio-org/fe-tl-reviewers , can someone please review this PR.

**Review by BogdanDenis — APPROVED** [review]

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js [hunk 5]
```diff
@@ -23,26 +23,37 @@ const ExternalLinkModal = ({
...
-      setError(false);
     } else {
       setDisabled(true);
     }
   }, [inputValue]);
 
-  const handleSave = () => {
+  const handleSave = async () => {
+    setExternalURLValidityError(null);
+    if (!inputValue) return;
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js:43]: This code is never hit as when there is no input, save button is disabled.

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js (hunk 5 above)
Comment (manvendra-s-rathore) [reply src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js]: yes, it is not needed now, removed.

**Review by Terala-Priyanka — APPROVED** [review]
